x/auction/client/common/query.go

```diff
@@ -0,0 +1,92 @@
+package common
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+
+	"github.com/kava-labs/kava/x/auction/types"
+)
+
+const (
+	defaultPage  = 1
+	defaultLimit = 30
+)
+
+// QueryAuctionByID returns an auction from state if present or falls back to searching old blocks
+func QueryAuctionByID(cliCtx context.CLIContext, cdc *codec.Codec, queryRoute string, auctionID uint64) (types.Auction, int64, error) {
+	bz, err := cdc.MarshalJSON(types.NewQueryAuctionParams(auctionID))
+	if err != nil {
+		return nil, 0, err
+	}
+
+	res, height, err := cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", queryRoute, types.QueryGetAuction), bz)
+
+	if err == nil {
+		var auction types.Auction
+		cdc.MustUnmarshalJSON(res, &auction)
+
+		return auction, height, nil
+	}
+
+	// NOTE: !errors.Is(err, types.ErrUnknownProposal) does not work here
+	if err != nil && !strings.Contains(err.Error(), "auction not found") {
+		return nil, 0, err
+	}
+
+	res, height, err = cliCtx.QueryWithData(fmt.Sprintf("custom/%s/%s", queryRoute, types.QueryNextAuctionID), nil)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	var nextAuctionID uint64
+	cdc.MustUnmarshalJSON(res, &nextAuctionID)
+
+	if auctionID >= nextAuctionID {
+		return nil, 0, sdkerrors.Wrapf(types.ErrAuctionNotFound, "%d", auctionID)
+	}
+
+	events := []string{
+		fmt.Sprintf("%s.%s='%s'", sdk.EventTypeMessage, sdk.AttributeKeyAction, "place_bid"),
+		fmt.Sprintf("%s.%s='%s'", types.EventTypeAuctionBid, types.AttributeKeyAuctionID, []byte(fmt.Sprintf("%d", auctionID))),
+	}
+
+	searchResult, err := utils.QueryTxsByEvents(cliCtx, events, defaultPage, defaultLimit)
```

Comment:
Does `QueryTxsByEvents` limit the returned results to `defaultLimit`?
If not a comment might be nice noting all txs are returned. If results are limited then this will miss auctions once we get more than 30 place bid msgs.

Reply:
Updated the limit to 100 and added a comment that the results will not capture the ending auction state if there are >100 bids.